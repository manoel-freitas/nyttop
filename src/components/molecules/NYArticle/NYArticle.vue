<template>
  <article class="media">
    <div class="media-content">
      <div class="content">
        <figure class="image" @click="$emit('onChooseArticle')">
          <img
            :src="image.url"
            :alt="image.caption"
            :width="image.width"
            :height="image.height"
          />
        </figure>
        <p>
          <strong style="max-width: 50px" @click="$emit('onChooseArticle')">
            {{ article.title }}
          </strong>
          <br />
          <small class="has-text-grey is-size-7">
            {{ article.published_date }}
            {{ article.byline }}
          </small>
          <br />
          <span class="is-size-6">{{ article.abstract }}</span>
        </p>
        <p v-if="article.kicker">
          <span class="is-size-6">{{ article.kicker }}</span>
        </p>
      </div>
    </div>
  </article>
</template>

<script>
export default {
  props: {
    article: {
      type: Object,
      required: true
    },
    image: {
      type: Object,
      required: true
    }
  }
}
</script>

<style lang="scss" scoped>
.content figure:not(:last-child) {
  margin-bottom: 0;
}
</style>
