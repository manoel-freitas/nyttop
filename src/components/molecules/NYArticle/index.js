import NYArticle from './NYArticle'

export default NYArticle
