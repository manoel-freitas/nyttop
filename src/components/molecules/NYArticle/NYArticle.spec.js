/// <reference types="jest" />

import { mount } from '@vue/test-utils'
import NYArticle from './NYArticle'

const requiredProps = {
  article: {
    image: {
      url: '',
      caption: '',
      width: '',
      heigth: ''
    }
  },
  image: {
    url: '',
    caption: '',
    width: '',
    heigth: ''
  }
}
const nyArticle = mount(NYArticle, {
  propsData: requiredProps
})

test('NYArticle has to be mounted', () => {
  expect(nyArticle.isVueInstance()).toBeTruthy()
})

test('snapshot test', () => {
  expect(nyArticle.element).toMatchSnapshot()
})
