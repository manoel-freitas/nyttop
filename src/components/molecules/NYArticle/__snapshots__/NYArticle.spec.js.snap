// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<article
  class="media"
>
  <div
    class="media-content"
  >
    <div
      class="content"
    >
      <figure
        class="image"
      >
        <img
          alt=""
          src=""
          width=""
        />
      </figure>
       
      <p>
        <strong
          style="max-width: 50px;"
        >
          
          
        
        </strong>
         
        <br />
         
        <small
          class="has-text-grey is-size-7"
        >
          
          
          
        
        </small>
         
        <br />
         
        <span
          class="is-size-6"
        >
          
        </span>
      </p>
       
      <!---->
    </div>
  </div>
</article>
`;
